app/scripts/directives/unbindService.js

```diff
@@ -0,0 +1,121 @@
+    //     deploymentConfigs = _.toArray(deploymentConfigData.by('metadata.name'));
+    //     ctrl.applications = sortApplications(deploymentConfigs, deployments, replicationControllers, replicaSets, statefulSets);
+    //   });
+    //   DataService.list('replicationcontrollers', context).then(function(replicationControllerData) {
+    //     replicationControllers = _.reject(replicationControllerData.by('metadata.name'), $filter('hasDeploymentConfig'));
+    //     ctrl.applications = sortApplications(deploymentConfigs, deployments, replicationControllers, replicaSets, statefulSets);
+    //   });
+    //   DataService.list({
+    //     group: 'extensions',
+    //     resource: 'deployments'
+    //   }, context).then(function(deploymentData) {
+    //     deployments = _.toArray(deploymentData.by('metadata.name'));
+    //     ctrl.applications = sortApplications(deploymentConfigs, deployments, replicationControllers, replicaSets, statefulSets);
+    //   });
+    //   DataService.list({
+    //     group: 'extensions',
+    //     resource: 'replicasets'
+    //   }, context).then(function(replicaSetData) {
+    //     replicaSets = _.reject(replicaSetData.by('metadata.name'), $filter('hasDeployment'));
+    //     ctrl.applications = sortApplications(deploymentConfigs, deployments, replicationControllers, replicaSets, statefulSets);
+    //   });
+    //   DataService.list({
+    //     group: 'apps',
+    //     resource: 'statefulsets'
+    //   }, context).then(function(statefulSetData) {
+    //     statefulSets = _.toArray(statefulSetData.by('metadata.name'));
+    //     ctrl.applications = sortApplications(deploymentConfigs, deployments, replicationControllers, replicaSets, statefulSets);
+    //   });
+    // };
+
+    var unbindService = function() {
+      DataService.delete({
+        group: 'servicecatalog.k8s.io',
+        resource: 'bindings'
+      }, ctrl.selectedBinding, context).then(_.noop, function(err) {
+        ctrl.error = err;
+      });
+    };
+
+    var showService = function() {
+      ctrl.nextTitle = 'Delete';
+    };
+
+    var showConfirm = function() {
+      ctrl.nextTitle = 'Close';
+      ctrl.wizardComplete = true;
+      unbindService();
+    };
+
+    ctrl.$onInit = function() {
+      // loadApps();
+      ctrl.displayName = getDisplayName(ctrl.target);
+      ctrl.steps = [{
+        id: 'service',
+        label: 'Service',
+        view: 'views/directives/bind-service/select-binding.html',
+        onShow: showService
+      }, {
+        id: 'confirmation',
+        label: 'Confirmation',
+        view: 'views/directives/bind-service/remove-binding-confirm.html',
+        onShow: showConfirm
+      }];
+
+      context = {
+        namespace: _.get(ctrl.target, 'metadata.namespace')
+      };
+    };
+
+    ctrl.appNameForBinding = function(bindingName) {
+      return _.get(_.first(_.get(ctrl.applications, bindingName)), ['metadata', 'name']) || bindingName;
```

Comment:
It looks like we're showing just the first app name? Usually there is only one app, but if unbind is going to affect multiple apps, it's important to know.